shared/mempool/tx_mempool.go

```diff
@@ -0,0 +1,15 @@
+package mempool
+
+// type GenericFIFOList = mempool_list.GenericFIFOList[any] // TODO: fix this aliasing
```

Comment:
Why do we need/want to create these types?

Reply:
I was trying to be fancy. Basically `GenericFIFOList` resides under the `list` folder 
![image](https://user-images.githubusercontent.com/29378614/216633210-f211c3dd-9e63-445d-8d10-f1ce83ae3f99.png)
and when importing we need to do this:
![image](https://user-images.githubusercontent.com/29378614/216633320-7a218e05-43a5-4b9c-9889-73ca2e333ebe.png)

I wanted to alias it so that it was coming from the root package but with generics it's not possible it seems, I haven't spent more than 10secs on it.

Reply:
I have removed that comment in the meantime BTW